src/auto_code_show_messages_and_followups.py

```diff
@@ -0,0 +1,105 @@
+import random
+import time
+from os import path
+
+from core_data_modules.cleaners import somali
+from core_data_modules.logging import Logger
+from core_data_modules.traced_data import Metadata
+from core_data_modules.traced_data.io import TracedDataCSVIO, TracedDataCodaV2IO
+from core_data_modules.util import IOUtils
+
+from src.lib import PipelineConfiguration, MessageFilters, ICRTools
+#TODO Uncomment once channels.py is ready
+#from src.lib.channels import Channels
+
+log = Logger(__name__)
+
+class AutoCodeShowMessagesAndFollowups(object):
+    RQA_AND_FOLLOW_UP_KEYS = []
+    for plan in PipelineConfiguration.RQA_CODING_PLANS + PipelineConfiguration.FOLLOW_UP_CODING_PLANS:
+        RQA_AND_FOLLOW_UP_KEYS.append(plan.raw_field)
+
+    SENT_ON_KEY = "sent_on"
+    NOISE_KEY = "noise"
+    ICR_MESSAGES_COUNT = 200
+    ICR_SEED = 0
+
+    @classmethod
+    
+    def auto_code_show_messages_and_followups(cls, user, data, icr_output_dir, coda_output_dir):
+        # Filter out test messages sent by AVF.
+        if not PipelineConfiguration.DEV_MODE:
+            data = MessageFilters.filter_test_messages(data)
+        
+        # Filter out runs sent outwith the project start and end dates
+        data = MessageFilters.filter_time_range(
+            data, cls.SENT_ON_KEY, PipelineConfiguration.PROJECT_START_DATE, PipelineConfiguration.PROJECT_END_DATE)
+        
+        # Filter for runs which don't contain a response to any week's question
+        data = MessageFilters.filter_empty_messages(data, cls.RQA_AND_FOLLOW_UP_KEYS)
+
+        # Tag RQA messages which are noise as being noise
+        for td in data:
+            is_noise = True
+            for rqa_key in cls.RQA_AND_FOLLOW_UP_KEYS:
+                if rqa_key in td and not somali.DemographicCleaner.is_noise(td[rqa_key], min_length=10):
+                    is_noise = False
+            td.append_data({cls.NOISE_KEY: is_noise}, Metadata(user, Metadata.get_call_location(), time.time()))
+        
+        #TODO Uncomment once channels.py is ready
+        #Label each message with channel keys
+        #Channels.set_channel_keys(user, data, cls.SENT_ON_KEY)
+
+        # Filter for messages which aren't noise (in order to export to Coda and export for ICR)
+        not_noise = MessageFilters.filter_noise(data, cls.NOISE_KEY, lambda x: x)
+
+        # Compute the number of messages that have empty string
+        log.debug("Counting the number of empty string messages for each raw field...")
```

Comment:
This doesn't work, see https://github.com/AfricasVoices/Project-ADSS/pull/52

Reply:
Thank you have fixed this.